run_e2e.sh

```diff
@@ -24,9 +24,20 @@ shutdown () {
 # We get the TERM from kubernetes and handle it gracefully
 trap shutdown TERM
 
-echo "/usr/local/bin/e2e.test --disable-log-dump --repo-root=/kubernetes --ginkgo.skip=\"${E2E_SKIP}\" --ginkgo.focus=\"${E2E_FOCUS}\" --provider=\"${E2E_PROVIDER}\" --report-dir=\"${RESULTS_DIR}\" --kubeconfig=\"${KUBECONFIG}\" --ginkgo.noColor=true"
-/usr/local/bin/e2e.test --disable-log-dump --repo-root=/kubernetes --ginkgo.skip="${E2E_SKIP}" --ginkgo.focus="${E2E_FOCUS}" --provider="${E2E_PROVIDER}" --report-dir="${RESULTS_DIR}" --kubeconfig="${KUBECONFIG}" --ginkgo.noColor=true | tee ${RESULTS_DIR}/e2e.log &
-# $! is the pid of tee, not e2e.test
+ginkgo_args=(
+    "--focus=${E2E_FOCUS}"
+    "--skip=${E2E_SKIP}"
+    "--noColor=true"
+)
+
+case ${E2E_PARALLEL} in
+    'y'|'Y')           ginkgo_args+=("--nodes=25") ;;
+    [1-9]|[1-9][0-9]*) ginkgo_args+=("--nodes=${E2E_PARALLEL}") ;;
+esac
+
+echo "/usr/local/bin/ginkgo ${ginkgo_args[@]} /usr/local/bin/e2e.test -- --disable-log-dump --repo-root=/kubernetes --provider=\"${E2E_PROVIDER}\" --report-dir=\"${RESULTS_DIR}\" --kubeconfig=\"${KUBECONFIG}\""
```

Comment:
The way the e2e test is smashed you don't need to run ginkgo this way the binary is self contained, did you crib this from upstream?

Reply:
This is what ginkgo wrapper uses afaik. I couldn't work out a way to use these without going through ginkgo, but if you know differently, please let me know how! Ginkgo spawns `node` number of processes to run tests in parallel, the `e2e.test` binary doesn't do that afaik.

Reply:
FYI - running

```
$ ./e2e.test --kubeconfig=<PATH_TO_KUBECONFIG> --ginkgo.parallel.total=25 --ginkgo.focus="Pods should be submitted and removed"
```

Just hangs, saying this is node 1 but there being nowhere to report back to.